### PR TITLE
[fix] fix the `@@` prefix issue if `bazel.module` and `WORKSPACE` are used together

### PR DESCRIPTION
**Pull request by tgodzik**
Second attempt, this time I changed it to looks for the actual @@ or @ label to be available.

### CODE REVIEW

**Comment by tgodzik** [comment]
Ok, so this should be much safer. I don't do any assumptions here about @ or @@. Instead I decided to allow both as both are possible within a single workspace.

File: bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt [hunk 1]
```diff
@@ -16,12 +16,16 @@ data class BazelRelease(
...
-    else ->
-      if (isBzlModEnabled) "@@//"
-      else "@//"
+    in 4..5 -> label.startsWith("//")
+    else -> label.startsWith("@//") || label.startsWith("@@//")
+  }
+
+  fun stripPrefix(label: String) = when (major) {
+    in 0..3 -> throw RuntimeException("Unsupported Bazel version, use Bazel 4 or newer")
+    in 4..5 -> label.substring("//".length)
```
Comment (abrams27) [inline bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt:27]: i think it's better to use `removePrefix` here

File: bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt [hunk 2]
```diff
@@ -16,12 +16,16 @@ data class BazelRelease(
...
+    else -> label.dropWhile { it == '@' }.substring("//".length)
```
Comment (abrams27) [inline bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt:28]: and here ofc as well

File: server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt [hunk 3]
```diff
@@ -55,20 +56,20 @@ class ProjectResolver(
...
+    ) { buildAspectResult.bepOutput.filesByOutputGroupNameTransitive(BSP_INFO_OUTPUT_GROUP) }
+    val targets = measured(
+        "Parsing aspect outputs"
+    ) { targetInfoReader.readTargetMapFromAspectOutputs(aspectOutputs).let { it } }
     val allTargetNames =
       if (buildAspectResult.isFailure)
         measured(
           "Fetching all possible target names"
-        ) { formatTargetsIfNeeded(bazelBspFallbackAspectsManager.getAllPossibleTargets(cancelChecker)) }
+        ) { formatTargetsIfNeeded(bazelBspFallbackAspectsManager.getAllPossibleTargets(cancelChecker), targets) }
```
Comment (abrams27) [inline server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt:69]: note: targets from`bazelBspFallbackAspectsManager.getAllPossibleTargets` are not included in `targets so they will be always mapped to `@<target>`, but i think it's not a problem

**Review by abrams27 — COMMENTED** [review]
just minor comments, but seems to work on our project

File: bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt (hunk 1 above)
Comment (tgodzik) [reply bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt]: Fixed! I couldn't find the method before.

File: bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt (hunk 2 above)
Comment (tgodzik) [reply bazelrunner/src/main/kotlin/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt]: Done

File: server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt (hunk 3 above)
Comment (tgodzik) [reply server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ProjectResolver.kt:69]: That should work correctly as far as I can see :thinking:

**Comment by tgodzik** [comment]
I confirmed it just in case on the private repository I am working on and it seems to work alright. I also rebased the PR, which seems to have made the CI issue go away.

**Comment by tgodzik** [comment]
Thanks for taking a look at it!

**Comment by tgodzik** [comment]
We could close https://youtrack.jetbrains.com/issue/BAZEL-863/Bazel-7.x-will-always-try-to-create-module.bazel

**Pull request merged**